test/runtime/fqdn.go

```diff
@@ -778,14 +782,94 @@ INITSYSTEM=SYSTEMD`
 		})
 	})
 
+	It("DNS proxy policy works if Cilium stops", func() {
+		targetURL := "http://world1.cilium.test"
+		targetIP := worldIps[WorldHttpd1]
+		invalidURL := "http://world1.outside.test"
+		invalidIP := outsideIps[OutsideHttpd1]
+
+		policy := `
+[
+	{
+		"labels": [{
+			"key": "dns-proxy"
+		}],
+		"endpointSelector": {
+			"matchLabels": {
+				"container:id.app1": ""
+			}
+		},
+		"egress": [
+			{
+				"toPorts": [{
+					"ports":[{"port": "53", "protocol": "ANY"}],
+					"rules": {
+						"dns": [
+							{"matchPattern": "*.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [{
+					"matchName": "world1.cilium.test"
+				}]
+			}
+		]
+	}
+]`
+		_, err := vm.PolicyRenderAndImport(policy)
+		Expect(err).To(BeNil(), "Policy cannot be imported")
+
+		expectFQDNSareApplied()
+
+		By("Curl from %q to %q", helpers.App1, targetURL)
+		res := vm.ContainerExec(helpers.App1, helpers.CurlFail(targetURL))
+		res.ExpectSuccess("Cannot connect from app1")
+
+		By("Curl from %q to %q should fail", helpers.App1, invalidURL)
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(invalidURL))
+		res.ExpectFail("Can connect from app1 when it should not work")
+
+		By("Stopping Cilium")
+
+		defer func() {
+			// Defer a Cilium restart to make sure that keep started when test finished.
+			_ = vm.ExecWithSudo("systemctl start cilium")
+			vm.WaitEndpointsReady()
+		}()
+
+		res = vm.ExecWithSudo("systemctl stop cilium")
+		res.ExpectSuccess("Failed trying to stop cilium via systemctl")
+
+		By("Testing connectivity to the IP %q without DNS request", targetIP)
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail("http://%s", targetIP))
+		res.ExpectSuccess("Cannot connect app1")
+
+		By("Curl from %q to %q with Cilium down", helpers.App1, targetURL)
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(targetURL))
+		res.ExpectFail("This request should fail because no dns-proxy")
```

Comment:
Hi, 

There is a DNS proxy in there, so the DNS will not work. IP connectivity is allow because the CIDR is in there, but the DNS request is redirected to proxy that it's not running.

Reply:
It might be worth putting more info in the fail message, for this specific check:"This request is expected to fail because the DNS Proxy in cilium-agent is not available when the agent is down".
This is so that when we "fix" this it'll be obvious that the test also needs to be changed. The reason to include the check now is mostly to sanity check that the DNS lookups are going through the proxy, and not accidentally bypassing it.

Reply:
commend added